test/extended/builds/pipeline.go

```diff
@@ -50,4 +58,133 @@ var _ = g.Describe("[builds][Slow] openshift pipeline build", func() {
 		})
 	})
 
+	g.Context("Pipeline with maven slave", func() {
+		g.It("Should build and complete successfully", func() {
+			// instantiate the template
+			g.By(fmt.Sprintf("calling oc new-app -f %q", mavenSlavePipelinePath))
+			err := oc.Run("new-app").Args("-f", mavenSlavePipelinePath).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			// start the build
+			g.By("starting the pipeline build and waiting for it to complete")
+			br, _ := exutil.StartBuildAndWait(oc, "openshift-jee-sample")
+			br.AssertSuccess()
+
+			// wait for the service to be running
+			g.By("expecting the openshift-jee-sample service to be deployed and running")
+			_, err = exutil.GetEndpointAddress(oc, "openshift-jee-sample")
+			o.Expect(err).NotTo(o.HaveOccurred())
+		})
+	})
+
+	g.Context("Orchestration pipeline", func() {
+		g.It("Should build and complete successfully", func() {
+			// instantiate the template
+			g.By(fmt.Sprintf("calling oc new-app -f %q", orchestrationPipelinePath))
+			err := oc.Run("new-app").Args("-f", orchestrationPipelinePath).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			// start the build
+			g.By("starting the pipeline build and waiting for it to complete")
+			br, _ := exutil.StartBuildAndWait(oc, "mapsapp-pipeline")
+			br.AssertSuccess()
+
+			// wait for the service to be running
+			g.By("expecting the parksmap-web service to be deployed and running")
+			_, err = exutil.GetEndpointAddress(oc, "parksmap")
+			o.Expect(err).NotTo(o.HaveOccurred())
+		})
+	})
+
+	g.Context("Blue-green pipeline", func() {
+		g.It("Should build and complete successfully", func() {
+			// Deploy Jenkins without oauth
+			g.By(fmt.Sprintf("calling oc new-app -f %q -p ENABLE_OAUTH=false", jenkinsTemplatePath))
+			err := oc.Run("new-app").Args("-f", jenkinsTemplatePath, "-p", "ENABLE_OAUTH=false").Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			// Wait for Jenkins to deploy
+			g.By("waiting for Jenkins to deploy")
+			err = exutil.WaitForADeploymentToComplete(oc.KubeClient().Core().ReplicationControllers(oc.Namespace()), "jenkins", oc)
```

Comment:
you don't wait for the jenkins deployment in the other tests, why do you need to wait for it here?